windows/RNSVG/ImageView.cpp

```diff
@@ -96,52 +113,61 @@ void ImageView::Render(UI::Xaml::CanvasControl const &canvas, CanvasDrawingSessi
     height = m_source.height * m_source.scale;
   }
 
-  Effects::Transform2DEffect transformEffect{nullptr};
-  if (m_align != "") {
-    Rect elRect{x, y, width, height};
-    Rect vbRect{0, 0, m_source.width, m_source.height};
-    transformEffect = Effects::Transform2DEffect{};
-    transformEffect.TransformMatrix(Utils::GetViewBoxTransform(vbRect, elRect, m_align, m_meetOrSlice));
+  com_ptr<ID2D1Geometry> clipPathGeometry;
+  if (ClipPathGeometry()) {
+    clipPathGeometry = get_self<D2DGeometry>(ClipPathGeometry())->Get();
   }
 
-  auto const &clipPathGeometry{ClipPathGeometry()};
+  D2DHelpers::PushOpacityLayer(deviceContext.get(), clipPathGeometry.get(), m_opacity);
 
-  if (auto const &opacityLayer{clipPathGeometry ? session.CreateLayer(m_opacity, clipPathGeometry) : session.CreateLayer(m_opacity)}) {
-    if (m_source.format == ImageSourceFormat::Bitmap && m_bitmap) {
-      auto const &transform{session.Transform()};
-      if (m_propSetMap[RNSVG::BaseProp::Matrix]) {
-        session.Transform(SvgTransform());
-      }
+  if (m_source.format == ImageSourceFormat::Bitmap) {
+    D2D1_MATRIX_3X2_F transform{D2DHelpers::GetTransform(deviceContext.get())};
 
-      if (m_align != "" && transformEffect) {
-        transformEffect.Source(m_bitmap);
-        Effects::CropEffect cropEffect{};
-        cropEffect.SourceRectangle({x, y, width, height});
-        cropEffect.Source(transformEffect);
-        session.DrawImage(cropEffect);
-      } else {
-        session.DrawImage(m_bitmap, {x, y, width, height});
-      }
+    if (m_propSetMap[RNSVG::BaseProp::Matrix]) {
+      deviceContext->SetTransform(transform * D2DHelpers::AsD2DTransform(SvgTransform()));
+    }
 
-      session.Transform(transform);
+    if (m_align != "") {
+      com_ptr<ID2D1Effect> bitmapEffects;
+      check_hresult(deviceContext->CreateEffect(CLSID_D2D1BitmapSource, bitmapEffects.put()));
+      check_hresult(bitmapEffects->SetValue(D2D1_BITMAPSOURCE_PROP_WIC_BITMAP_SOURCE, m_wicbitmap.get()));
+
+      com_ptr<ID2D1Effect> transformEffect;
+      Rect elRect{x, y, width, height};
+      Rect vbRect{0, 0, m_source.width, m_source.height};
```

Comment:
What are elRect and vbRext standing for?

Reply:
elRect is for the element and vbRect for the viewbox. 

You can read more about viewbox transforms here:
https://svgwg.org/svg2-draft/coords.html#ComputingAViewportsTransform